File: Java/228. Summary Ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        int n = nums.length;
        List<String> res = new ArrayList();
        for(int i = 0; i < n; i++){
            int j = i;
            while(j + 1 < n && nums[j + 1] == nums[j] + 1){
                j++;
            }
            if(j > i){
                res.add(String.valueOf(nums[i]) + "->" + String.valueOf(nums[j]));
            }
            else{
                res.add(String.valueOf(nums[i]));
            }
            i = j;
        }
        return res;
    }
}

File: JavaScript/228. Summary Ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
    var n = nums.length;
    var res = new Array();
    for(var i = 0; i < n; i++){
        var j = i;
        while(j + 1 < n && nums[j + 1] == nums[j] + 1){
            j++;
        }
        if(j > i){
            res.push(nums[i].toString() + "->" + nums[j].toString());
        }
        else{
            res.push(nums[i].toString());
        }
        i = j;
    }
    return res;
};
